src/modules/task-list/task-list.ts

```diff
@@ -161,14 +167,33 @@ export class TaskList {
         }
       });
 
-    const userTaskListArray: Array<Array<IUserTaskWithProcessModel>> = await Promise.all(promisesForAllUserTasks);
+    const promisesForAllManualTasks: Array<Promise<Array<IManualTaskWithProcessModel>>> = allProcessModels.processModels
+      .map(async(processModel: ProcessModelExecution.ProcessModel): Promise<Array<IManualTaskWithProcessModel>> => {
+        try {
+          const manualTaskList: ManualTaskList = await this._managementApiService.getManualTasksForProcessModel(identity, processModel.id);
+
+          const manualTasksAndProcessModels: Array<IManualTaskWithProcessModel> = this._addProcessModelToManualTasks(manualTaskList, processModel);
 
-    const flattenedUserTasks: Array<IUserTaskWithProcessModel> = [].concat(...userTaskListArray);
+          return manualTasksAndProcessModels;
 
-    return flattenedUserTasks;
+        } catch (error) {
+          if (isError(error, NotFoundError)) {
+            // the management api returns a 404 if there is no instance of a process model running.
+            return Promise.resolve([]);
+          }
+          throw error;
+        }
+     });
+
+    const fullPromise: Array<Promise<Array<IUserTaskWithProcessModel & IManualTaskWithProcessModel>>> = []
```

Comment:
```suggestion
    type PromisesForUserAndManualTasks = Promise<Array<IUserTaskWithProcessModel & IManualTaskWithProcessModel>>
    type  ListOfPromisesForUserAndManualTasks = Array<PromisesForUserAndManualTasks>;

    const promiseList: PromiseListForUserAndManualTasks = []
```

@SteffenKn What about this?

Then use the types afterwards.